### PR TITLE
Use v1beta1 API in E2E tests

### PR DESCRIPTION
**Pull request by xmudrii**
**What this PR does / why we need it**:

Use the v1beta1 API for E2E tests.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
xref #828 

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/assign @kron4eg

### CODE REVIEW

File: test/e2e/kubeone.go [hunk 1]
```diff
@@ -51,25 +51,46 @@ func NewKubeone(kubeoneDir, configurationFilePath string) *Kubeone {
...
+	providerName string,
 	providerExternal bool,
 	clusterNetworkPod string,
 	clusterNetworkService string,
 	credentialsFile string,
 ) error {
 	k1Cluster := k1api.KubeOneCluster{
 		TypeMeta: metav1.TypeMeta{
-			APIVersion: "kubeone.io/v1alpha1",
+			APIVersion: "kubeone.io/v1beta1",
```
Comment (kron4eg) [inline test/e2e/kubeone.go:62]:
```suggestion
			APIVersion: k1api.SchemeGroupVersion.String(),
```

File: test/e2e/kubeone.go [hunk 2]
```diff
@@ -51,25 +51,46 @@ func NewKubeone(kubeoneDir, configurationFilePath string) *Kubeone {
...
 		},
 	}
 
 	k1api.SetObjectDefaults_KubeOneCluster(&k1Cluster)
 
 	k1Cluster.CloudProvider = k1api.CloudProviderSpec{
-		Name:     providerName,
 		External: providerExternal,
 	}
+	switch providerName {
```
Comment (kron4eg) [inline test/e2e/kubeone.go:72]: in v1beta1 package we have private `func setCloudProvider(cp *kubeonev1beta1.CloudProviderSpec, name string) error {`, probably it's time to make it public.

Comment (xmudrii) [reply test/e2e/kubeone.go:72]: That function is in the `terraform/v1beta1` package. Should we move it to some other location or expose it from there?

Comment (kron4eg) [reply test/e2e/kubeone.go:72]: yeah, to the api/kubeone/v1beta1 probably

File: test/e2e/kubeone.go (hunk 1 above)
Comment (xmudrii) [reply test/e2e/kubeone.go:62]: Fixed

File: test/e2e/kubeone.go (hunk 2 above)
Comment (xmudrii) [reply test/e2e/kubeone.go]: Fixed

**Comment by xmudrii** [comment]
@kron4eg All comments addressed, PTAL

**Pull request merged**